src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -1,26 +1,48 @@
-﻿using System;
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Diagnostics.Monitoring.Contracts;
+using Microsoft.Diagnostics.Monitoring.Logging;
 using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Diagnostics.Monitoring
 {
     public sealed class DiagnosticServices : IDiagnosticServices
     {
-        private const int MaxTraceSeconds = 60 * 5;
-
         private readonly ILogger<DiagnosticsMonitor> _logger;
         private readonly CancellationTokenSource _tokenSource = new CancellationTokenSource();
-
-        public DiagnosticServices(ILogger<DiagnosticsMonitor> logger)
+        private IEnumerable<IMetricsLogger> _metricsLoggers;
+        private ContextConfiguration _contextConfiguration;
+        private IServiceProvider _loggerServiceProvider;
+
+        public DiagnosticServices(ILogger<DiagnosticsMonitor> logger,
+            IOptions<ContextConfiguration> contextConfig,
+            IEnumerable<IMetricsLogger> metricsLoggers,
+            IStreamAccessor streamingAccessor)
         {
             _logger = logger;
+            _metricsLoggers = metricsLoggers;
+            _contextConfiguration = contextConfig.Value;
+
+            //We want to use ILogger since at some point in the future we may want to push this data to other
+            //sinks. At the same time we want these to be transient for the purposes of streaming the data, but not for other logs (such as those of
+            //the rest service itself).
+            ServiceCollection serviceCollection = new ServiceCollection();
```

Comment:
This seems like a bad idea. Is this because we happen to be using the ILoggerProvider interface here? Can we just new this up when we need to for the specific call?

Reply:
I think you came to the same conclusion I did in my feedback : )

Reply:
We can certainly do that. I wanted to stick with ServiceCollection here to make it very straightforward to add other loggers using common conventions, but it did create a much more complicated initialization.

Reply:
My guess is that if we have other loggers later either it won't be within the context of the REST server (not created at this point in the code) or the REST server will have hard coded knowledge of those options too.